src/editor/Cloud.tsx

```diff
@@ -0,0 +1,49 @@
+import toast from "react-hot-toast"
+
+import { toGateway } from "~/lib/ipfs-parser"
+import { UploadFile } from "~/lib/upload-file"
+
+import { ICommand } from "."
+
+const uploadResources = async (text: string) => {
+  const markdownImageRegex = /!\[(.*)\]\((.+)\)/
```

Comment:
Links that are already in IPFS `![](ipfs://xxxx)` need to be excluded, so it is best to filter links that start with `https://`.

Reply:
> Links that are already in IPFS `![](ipfs://xxxx)` need to be excluded, so it is best to filter links that start with `https://`.

indeed. wait me to fix it later

Reply:
Resolved. filtered by startWith below

Reply:
You an be filtered out directly with the regular `/! \[. *? \]\((https:\/\/. *?) \)/`

Also this seems to only upload one image? I asked ChatGPT to write one

```js
// 定义 markdown 文本
let markdown = `# Title
Text with ![alt text](https://example.com/image1.jpg) image and ![another image](https://example.com/image2.png) in it.`;

// 定义匹配图片 URL 的正则表达式
let regex = /!\[.*?\]\((https:\/\/.*?)\)/g;

// 找到所有的匹配项
let matches = markdown.match(regex);

if (matches) {
    // 提取 URL
    let urls = matches.map(match => {
        return match.replace(/!\[.*?\]\((https:\/\/.*?)\)/, '$1');
    });

    console.log(urls); // 输出：[ 'https://example.com/image1.jpg', 'https://example.com/image2.png' ]
} else {
    console.log("No images found.");
}
```

Reply:
Exactly, there may be two images in a row. Let me revise it again.